### ISSUE TITLE
diagnostics.tidy out of sync with NeoVim

### ISSUE THREAD

**Issue by c3n21** [jose-elias-alvarez/null-ls.nvim#1040]
### FAQ

- [X] I have checked the [FAQ](https://github.com/jose-elias-alvarez/null-ls.nvim#faq) and it didn't resolve my problem.

### Issues

- [X] I have checked [existing issues](https://github.com/jose-elias-alvarez/null-ls.nvim/issues?q=is%3Aissue) and there are no issues with the same problem.

### Neovim Version

NVIM v0.8.0-dev-914-gd4d27c41b

### Operating System

Linux winnie 5.10.102.1-microsoft-standard-WSL2 #1 SMP Wed Mar 2 00:30:59 UTC 2022 x86_64 GNU/Linux

### Minimal config

```lua
local null_ls = require('null-ls')
-- register any number of sources simultaneously
local sources = {
    null_ls.builtins.diagnostics.tidy,
}

null_ls.setup({
    cmd = { 'nvim' },
    debounce = 250,
    debug = true,
    default_timeout = 1000,
    diagnostics_format = "[#{c}] #{m} (#{s})",
    fallback_severity = vim.diagnostic.severity.ERROR,
    log = {
        enable = true,
        level = 'warn',
        use_console = 'async',
    },
    on_attach = nil,
    -- function(client, bufnr)
    --     if client.supports_method('textDocument/formatting') then
    --         vim.api.nvim_clear_autocmds({ group = augroup, buffer = bufnr })
    --         vim.api.nvim_create_autocmd('BufWritePre', {
    --             group = augroup,
    --             buffer = bufnr,
    --             callback = function()
    --                 -- on 0.8, you should use vim.lsp.buf.format({ bufnr = bufnr }) instead
    --                 vim.lsp.buf.format({ bufnr = bufnr })
    --             end,
    --         })
    --     end
    -- end,
    on_init = nil,
    on_exit = nil,
    --[[ root_dir = u.root_pattern(".null-ls-root", "Makefile", ".git"), ]]
    update_in_insert = false,
    sources = sources,
})

```

### Steps to reproduce

Every time an error is made in the XML, the error isn't shown right away, but instead it needs a change in the buffer, whether is typing a character or undoing with 'u'

https://user-images.githubusercontent.com/37077738/186261426-ba160a76-3070-4b31-befa-b389c30b21f1.mp4

### Expected behavior

Diagnostics should be shown right away

### Actual behavior

Diagnostics is shown only after the buffer has changed one more time

### Debug log

[null-ls.log](https://github.com/jose-elias-alvarez/null-ls.nvim/files/9410527/null-ls.log)

### Help

Yes, but I don't know how to start. I would need guidance

### Requirements

- [X] I have read and followed the instructions above and understand that my issue will be closed if I did not provide the required information.

**Comment by jose-elias-alvarez** [comment]
So the issue is that our current implementation is reading the file from the disk, not `stdin`, which is making diagnostics go out of sync. According to `man tidy`, leaving out the filename will make the command read from `stdin`, but it seems to break the output format. This is a quick and dirty fix:

```diff
diff --git a/lua/null-ls/builtins/diagnostics/tidy.lua b/lua/null-ls/builtins/diagnostics/tidy.lua
index 8234a2b..78cc00f 100644
--- a/lua/null-ls/builtins/diagnostics/tidy.lua
+++ b/lua/null-ls/builtins/diagnostics/tidy.lua
@@ -21,11 +21,8 @@ return h.make_builtin({
         command = "tidy",
         args = function(params)
             local common_args = {
-                "--gnu-emacs",
-                "yes",
                 "-quiet",
                 "-errors",
-                "$FILENAME",
             }
 
             if params.ft == "xml" then
@@ -41,8 +38,8 @@ return h.make_builtin({
             return code <= 2
         end,
         on_output = h.diagnostics.from_pattern(
-            [[([^:]+):(%d+):(%d+): (%a+): (.+)]],
-            { "file", "row", "col", "severity", "message" },
+            [[line (%d+) column (%d+) %- (%a+): (.+)]],
+            { "row", "col", "severity", "message" },
             { severities = severities }
         ),
     },
```

@ObserverOfTime Do you have any ideas? Even if I leave in `--gnu-emacs yes` it seems to default to the regular output format when the filename isn't included.

**Comment by ObserverOfTime** [comment]
I guess the emacs format breaks because there's no filename. It's fine to use the default format in that case.

**Comment by c3n21** [comment]
For me it seems to work atm

**Comment by jose-elias-alvarez** [comment]
Thanks all! I pushed my quick fix in bf297db3956129183e889d58912735027a58d119. Feel free to open another issue / PR if it's behaving weirdly.
